modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/writer/ObjectWriterFactory.java

```diff
@@ -45,6 +50,31 @@ public static ObjectWriter getObjectWriter(List<String> includeFieldNames) {
 			enable(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY);
 			enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS);
 			setDateFormat(new ISO8601DateFormat());
+			registerModule(
+				new SimpleModule() {
+					{
+						addSerializer(
+							UnsafeSupplier.class,
+							new JsonSerializer<UnsafeSupplier>() {
```

Comment:
@albertojml can you extract this serializer to a separate class and reuse it in batch engine?